### DISCUSSION TITLE
게시판 이미지 처리에 대한 논의

### DISCUSSION THREAD

**Discussion by jjangsky** [two-clock/backend#75] [Ideas]
@openmpy 님 안녕하십니까
* #57 관련하여 논의가 필요하여 Discussions을 생성하였습니다.

게시판 API를 만들면서 게시글 이미지에 대한 처리를 어떠한 방식으로 할 지 의견이 필요할 것 같습니다.

첫 번째로  게시글의 Content 영역과 image 업로드 영역을 분리하여 게시글 저장 시 
이미지 파일도 업로드 되는 방식입니다.
단점이라고 보면 게시글 작성 과정에서 이미지 사이즈를 조절 한다거나 이미지 사이에 문장을 적는다거나
이러한 부분이 힘들 것 같습니다.

두 번째로는 웹 에디터를 사용하는 방식입니다.
`summernote` , `ckEditor` 등 에디터 모듈을 설치하여 사용하는 방법인데
장점으로는 글 작성 측면에서 자유도가 높은 것이 있고 이미지 파일 자체를 저장할 필요가 없는
것으로 알고 있습니다. -> base64로 인코딩 되어 바로 사용 가능

세 번째로는 깃허브의 issues나 pr 작성 시 드래그앤 드롭으로 이미지 가져가면 작성 화면에서 바로 서버로
업로드 되어 사용할 수 있는 방식입니다.
단점으로는 불필요한 이미지가 계속 업로드 될 수 있다는 점? 이 있겠군요

아마 대부분의 토이 프로젝트에서는 1번 방식으로 게시판을 개발하지 않을까 싶은데
@openmpy 님 의견이 필요하여 여쭙니다.

**Comment by openmpy** [comment]
@jjangsky 님 확인이 늦었습니다. 죄송합니다.
사실 두번째 방식이 제일 간편해 보이나.. 토이 프로젝트 목적을 생각해본다면 첫번째 방식으로 진행하는게 맞는거 같습니다.
S3 같은 스토리지 작업과 추후에 이미지 최적화 기능 같은걸 생각해보면 첫번째 방식으로 진행하는게 좀 더 유리해보입니다.
